libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/reposerver/ReposerverClient.scala

```diff
@@ -55,10 +53,13 @@ object ReposerverClient {
 trait ReposerverClient {
   protected def ReposerverError(msg: String) = RawError(ErrorCode("reposerver_remote_error"), StatusCodes.BadGateway, msg)
 
-  def createRoot(namespace: Namespace): Future[RepoId]
+  def createRoot(namespace: Namespace, keyType: KeyType): Future[RepoId]
 
   def fetchRoot(namespace: Namespace): Future[SignedPayload[RootRole]]
 
+  def repoExists(namespace: Namespace)(implicit ec: ExecutionContext): Future[Boolean] =
```

Comment:
why is this method needed? It can always be checked using fetchRoot.

This is catching other exceptions that should not be caught, should be something like

`.map(_ => true).recover { case RepoNotFound => false }`  or

`.transform { case Success(_) => Success(true) ; case Failure(RepoNotFound) => Success(false)}`

otherwise any exception will cause `false`

Reply:
Jerry wants to move the account setup code from app to user-profile and I wanted to use this repo client for it.

I'd say the above code does roughly what https://github.com/advancedtelematic/ota-plus-server/blob/a57aac4726d6726b25e0db770227b0c19e083d69/ota-plus-web/app/com/advancedtelematic/api/ApiClient.scala#L328 does (unless the HTTP client here would throw some low-level exception).

Reply:
you can still use the client to fetch the root and not add any method to it.

it's not the same, the code you link does not hide the any exception.

Reply:
Us having this discussions means it's a good idea to have this method.

The method in app returns "false" for the status codes "Locked" and "Failed Dependency", is that by accident or on purpose?

Reply:
Ah ok, git makes it look like you wrote that code.

How would you write TufRepoCreation.exists() then? You'd still need to decide how to deal with "Locked" and "Failed Dependency".

I agree that swallowing all exceptions was a bad idea.

Reply:
Depends on what you are trying to do, the equivalent code from app, but without swalling exceptions would be:
```
  def rootExists(namespace: Namespace): Future[Boolean] =
    fetchRoot(namespace).map(_ => true).recover {
      case NotFound | KeysNotReady | RootNotInKeyserver => false
    }
```

But, looking at the code, looks like FailedDependency is never actually returned from the server. Locked means the root exists already, but is not ready, so a create request would fail, so probably this would make more sense:

```
  def rootExists(namespace: Namespace): Future[Boolean] =
    fetchRoot(namespace).map(_ => true).recover {
      case NotFound | RootNotInKeyserver => false
      case KeysNotReady => true
    }
```

Reply:
I am wrong, FailedDependency is thrown by reposerver, so needs to be caught in the recover block